supporting_scripts/generate_code_cov_table/generate_code_cov_table.py

```diff
@@ -99,17 +99,18 @@ def filter_files(file_names):
 def get_client_line_coverage(username, password, build_id, file_name):
     report_url = get_client_tests_cov_report_url(build_id)
     file_report_url = f"{report_url}/{file_name}.html"
-    logging.debug(f"GET {file_report_url}")
     response = requests.get(file_report_url, auth=(username, password))
+    logging.debug(f"GET {file_report_url} -> {response.status_code}")
 
-    soup = BeautifulSoup(response.content, "html.parser")
-    coverage_divs = soup.find_all("div", {"class": "fl pad1y space-right2"})
     line_coverage = None
-    if len(coverage_divs) >= 4:
-        line_coverage_strong = coverage_divs[3].find("span", {"class": "strong"})
-        line_coverage = line_coverage_strong.text.replace("%", "").strip()
+    if response.status_code == 200:
+        soup = BeautifulSoup(response.content, "html.parser")
+        coverage_divs = soup.find_all("div", {"class": "fl pad1y space-right2"})
+        if len(coverage_divs) >= 4:
+            line_coverage_strong = coverage_divs[3].find("span", {"class": "strong"})
+            line_coverage = line_coverage_strong.text.replace("%", "").strip()
+        logging.debug(f"Coverage for {file_name} -> line coverage: {line_coverage}")
```

Comment:
Maybe there should be an error log or something like this if the response is neither 200 nor 404?

Reply:
Thanks, added :)